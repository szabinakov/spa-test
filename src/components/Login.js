import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import axios from "axios";
import PropTypes from "prop-types";
import "../styles/Form.css";
import Alert from "./Alert.js";

export default function Login({ setUserState }) {
  const initialState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };
  const [alert, setAlert] = useState(initialState.alert);
  const [value, setValue] = useState();
  const history = useHistory();

  const handleChange = (event) => {
    setValue({
      ...value,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    axios
      .post("https://reqres.in/api/login", {
        email: value.email,
        password: value.password,
      })
      .then((response) => {
        setUserState({
          isUserLoggedIn: true,
        });
        history.push("/");
      })
      .catch(() => {
        setAlert({
          message: "Something went wrong! Please try again!",
          isSuccess: false,
        });
      });
  };

  return (
    <>
      <Alert message={alert.message} success={alert.isSuccess} />
      <div className="formContainer">
        <h1>Welcome to the App, Please login!</h1>
        <form className="form" onSubmit={handleSubmit}>
          <div>
            <input
              type="text"
              id="email"
              className="email"
              name="email"
              placeholder="Enter email!"
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <input
              type="password"
              id="password"
              className="password"
              name="password"
              placeholder="Enter password!"
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <button type="submit">Login</button>
          </div>
          <div>
            <Link to="/register">
              <p>Not a member? Register here!</p>
            </Link>
          </div>
        </form>
      </div>
    </>
  );
}

Login.propTypes = {
  setUserState: PropTypes.func,
};
