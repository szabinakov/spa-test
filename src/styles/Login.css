.loginFormContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 350px;
}

.loginForm {
  background-color: rgba(239, 71, 110);
  border-radius: 20px;
  box-shadow: 10px 10px 20px;
  padding: 10px;
  min-width: 260px;
}
.loginForm > div:first-child {
  margin-top: 0px;
}
.loginForm > div:last-child {
  margin-bottom: 0px;
}
.loginForm > div {
  display: flex;
  flex-direction: column;
  margin: 15px;
}

.loginForm > div > label {
  align-self: center;
  padding-bottom: 9px;
}

.loginForm > div > input {
  height: 20px;
  width: 80%;
  align-self: center;
  border-radius: 4px;
  box-shadow: 3px 3px 3px;
}

.loginForm > div > button {
  width: 60px;
  height: 25px;
  align-self: center;
  border-radius: 20px;
  text-transform: uppercase;
}
.loginForm > div > button:hover {
  text-decoration: none;
}

.loginForm > div > a > p {
  text-align: center;
  font-size: 10px;
}
