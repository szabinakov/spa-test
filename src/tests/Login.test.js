import { render } from "@testing-library/react";
import Login from "../components/Login.js";
import { BrowserRouter as Router } from "react-router-dom";

describe("Login", () => {
  const mocksetUserStateFunc = jest.fn();

  it("Matches the snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Login setUserState={mocksetUserStateFunc} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders every element", () => {
    const { getByTestId } = render(
      <Router>
        <Login setUserState={mocksetUserStateFunc} />
      </Router>
    );

    expect(getByTestId("form")).toHaveClass("form");
    expect(getByTestId("email")).toHaveClass("email");
    expect(getByTestId("password")).toHaveClass("password");
    expect(getByTestId("buttonDiv")).toHaveClass("buttonDiv");
    expect(getByTestId("link")).toHaveClass("link");
  });
});
