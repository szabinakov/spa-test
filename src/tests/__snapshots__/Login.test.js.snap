// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="formContainer"
  >
    <h1>
      Welcome to the App, Please login!
    </h1>
    <form
      class="form"
      data-testid="form"
    >
      <div>
        <input
          class="email"
          data-testid="email"
          id="email"
          name="email"
          placeholder="Enter email!"
          required=""
          type="text"
        />
      </div>
      <div>
        <input
          class="password"
          data-testid="password"
          id="password"
          name="password"
          placeholder="Enter password!"
          required=""
          type="password"
        />
      </div>
      <div
        class="buttonDiv"
        data-testid="buttonDiv"
      >
        <button
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        class="link"
        data-testid="link"
      >
        <a
          href="/register"
        >
          <p>
            Not a member? Register here!
          </p>
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
